### PR TITLE
Improve the documentation landing page

### PR DESCRIPTION
**Pull request by cclemonzy**
related with #195
# Changes
- change title from "Tekton Documentation" to "Welcome to Tekton"
- remove "What's next"
- add a list of 7 components with one sentence description about each
- all of the components name has link to its README.md on github
- add a diagram of pipeline

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/website/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

**Comment by cclemonzy** [comment]
Hi, @mpetason  thank you so much for your comment. I updated _index.md in PR #255.

**Comment by cclemonzy** [comment]
Hi, @afrittoli , @vdemeester and all.  Our PR for issue #255 now still in deploy failed stage, we have done  all what we could do. 
Do you have any ideas with this?   if any comment we will be very  appreciated

File: sync/config/README.md [hunk 1]
```diff
@@ -57,3 +57,28 @@ These two YAML files control the synchronization/curation from the `tektoncd/pip
...
+```yaml
+folders:
+  docs:
+    index: README.md
+    include: ['*.md']
+  docs/subdir:
+    target: subdir
+    index: README.md
+    include: ['*.md']
+```
```
Comment (afrittoli) [inline sync/config/README.md:84]: This is already merged in, it shouldn't be part of this PR?

File: sync/config/pipelines.yaml [hunk 2]
```diff
@@ -135,10 +135,10 @@ tags:
       - deprecations.md
       - install.md
       - tutorial.md
-- name: master
+- name: main
   # The name to display on tekton.dev.
   # sync.py will use this value in the version switcher and other places.
-  displayName: master
+  displayName: main
```
Comment (afrittoli) [inline sync/config/pipelines.yaml:141]: These are already in, they shouldn't be part of this PR

File: content/en/docs/_index.md [hunk 3]
```diff
@@ -1,20 +1,42 @@
...
+
+- **[Tekton Dashboard](https://github.com/tektoncd/dashboard/blob/main/README.md)** is a Web-based graphical interface for Tekton
+  Pipelines that displays information about the execution of your pipelines. It is currently a work-in-progress.
+
+- **[Tekton Catalog](https://github.com/tektoncd/catalog/blob/main/README.md)** is a repository of high-quality, community-contributed
+  Tekton building blocks - `Tasks`, `Pipelines`, and so on - that are ready for use in your own pipelines.
+
+- **[Tekton Hub](https://github.com/tektoncd/hub/blob/main/README.md)** is a Web-based graphical interface for accessing the Tekton Catalog.
+
+- **[Tekton Operator](https://github.com/tektoncd/operator/blob/main/README.md)** is a Kubernetes [Operator pattern](https://operatorhub.io/what-is-an-operator)
```
Comment (afrittoli) [inline content/en/docs/_index.md:33]: Perhaps replace "operator pattern" with "operator" ?

Comment (afrittoli) [reply content/en/docs/_index.md:33]: Tekton Results is missing from this list, but we could add it as a follow-up. @wlynch

File: content/en/docs/_index.md [hunk 4]
```diff
@@ -1,20 +1,42 @@
...
+- **[Tekton CLI](https://github.com/tektoncd/cli/blob/main/README.md)** provides a command-line interface called `tkn`, built on top
+  of the Kubernetes CLI, that allows you to interact with Tekton.
+
+- **[Tekton Dashboard](https://github.com/tektoncd/dashboard/blob/main/README.md)** is a Web-based graphical interface for Tekton
+  Pipelines that displays information about the execution of your pipelines. It is currently a work-in-progress.
+
+- **[Tekton Catalog](https://github.com/tektoncd/catalog/blob/main/README.md)** is a repository of high-quality, community-contributed
+  Tekton building blocks - `Tasks`, `Pipelines`, and so on - that are ready for use in your own pipelines.
+
+- **[Tekton Hub](https://github.com/tektoncd/hub/blob/main/README.md)** is a Web-based graphical interface for accessing the Tekton Catalog.
```
Comment (afrittoli) [inline content/en/docs/_index.md:31]: Perhaps we could use the first line from the linked README? 
> The Tekton hub is a web based platform for developers to discover, share and contribute tasks and pipelines for Tekton.

**Review by afrittoli — COMMENTED** [review]
Thanks for working on this, this is definitely an improvement in the right direction!

File: content/en/docs/_index.md [hunk 5]
```diff
@@ -1,20 +1,42 @@
...
 
-Tekton is a cloud native continuous integration and delivery (CI/CD)
-solution. It allows developers to build, test, and deploy across cloud
-providers and on-premise systems.
+Tekton is an open-source cloud native CI/CD (Continuous Integration and
+Delivery/Deployment) solution. It allows developers to build, test, and
+deploy across destinations using a Kubernetes cluster of their own.
 
-### What's next
+The Tekton project consists of the following components:
```
Comment (sergetron) [inline content/en/docs/_index.md:15]: We already had this documented very well in https://github.com/tektoncd/website/pull/168, but now I cannot find this file. What ever happened to it?

**Comment by sergetron** [comment]
I wrote a detailed overview of Tekton back in September 2020 in https://github.com/tektoncd/website/pull/168 but now I cannot find it anymore. What ever happened to it? I think that overview would work perfectly on this page.

File: content/en/docs/_index.md (hunk 5 above)
Comment (AlanGreene) [reply content/en/docs/_index.md:15]: Looks like it was moved in https://github.com/tektoncd/website/pull/214
Now it's at https://github.com/tektoncd/website/blob/2b51b90d0a4720c2c5d6a02b8c6ec8b1c8ce64ef/content/en/docs/Overview/_index.md

File: content/en/docs/_index.md (hunk 4 above)
Comment (AlanGreene) [reply content/en/docs/_index.md:31]: Looks like this content is largely a duplicate of the list from the [Overview page](https://tekton.dev/docs/overview/#what-are-the-components-of-tekton) which is already linked from the documentation landing page. We should either try to consolidate the two pages or keep a clear separation of the content.

File: content/en/docs/_index.md [hunk 6]
```diff
@@ -1,20 +1,42 @@
...
 weight: 20
 menu:
   main:
     weight: 20
 ---
 
-Tekton is a cloud native continuous integration and delivery (CI/CD)
-solution. It allows developers to build, test, and deploy across cloud
-providers and on-premise systems.
+Tekton is an open-source cloud native CI/CD (Continuous Integration and
```
Comment (sergetron) [inline content/en/docs/_index.md:11]: Typically the first time you introduce the name of a product, system, or feature, you write out the full name of the feature first, then include the acronym in parentheses, and subsequently only use the acronym. Not sure this change is necessary here.

File: content/en/docs/_index.md (hunk 5 above)
Comment (sergetron) [reply content/en/docs/_index.md:15]: Then it's probably best to link to the Overview instead of rehashing this content in slightly different wording.

Comment (AlanGreene) [reply content/en/docs/_index.md:15]: Agreed, duplication makes it harder to keep things consistent and up to date. The overview is already linked from the current landing page: 
image

File: sync/config/pipelines.yaml (hunk 2 above)
Comment (cclemonzy) [reply sync/config/pipelines.yaml:141]: thank you for your review. to reslove the deploy failure I did this merge to our PR. but it seems not right. I will resolve this with my fellow.

File: content/en/docs/_index.md (hunk 4 above)
Comment (cclemonzy) [reply content/en/docs/_index.md:31]: aggree with your comment, and according to comment from sergetron, a more complete description has been updated but not in main branch now. Maybe I should found it out and  reflect it in this PR?

File: sync/config/pipelines.yaml (hunk 2 above)
Comment (cclemonzy) [reply sync/config/pipelines.yaml:141]: Hi @afrittoli , To remove these excess content, I create another PR #263 with  all modifications in landing page Only. Could you please give a review? I will close this PR today.

File: content/en/docs/_index.md (hunk 3 above)
Comment (cclemonzy) [reply content/en/docs/_index.md:33]: @afrittoli  @wlynch  I think you mean the Tekton Results is required here. so I added it at PR #263. Could you please give a review? All modifications here are moved to #263,  I will close this PR today.

**Comment by cclemonzy** [comment]
@afrittoli  @AlanGreene @sergetron  Thank you  for all your comment here. And to remove excess commit content, I have created a new PR #263,  Could you please give a review?  Thanks in advance to all.   
Since all requirements in issue #195 are included in PR #263 ,  I will close this PR today.
